internal/run/workflowTracking.go

```diff
@@ -0,0 +1,261 @@
+package run
+
+import (
+	"fmt"
+	"github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/speakeasy-api/speakeasy/internal/styles"
+	"os"
+	"strings"
+)
+
+type Status string
+
+const (
+	StatusRunning   Status = "running"
+	StatusFailed    Status = "failed"
+	StatusSucceeded Status = "success"
+)
+
+type WorkflowStep struct {
+	name     string
+	status   Status
+	substeps []*WorkflowStep
+	nextStep *WorkflowStep
+	updates  chan<- UpdateMsg
+}
+
+func NewWorkflowStep(name string, sub chan<- UpdateMsg) *WorkflowStep {
+	return &WorkflowStep{
+		name:     name,
+		status:   StatusRunning,
+		substeps: []*WorkflowStep{},
+		updates:  sub,
+	}
+}
+
+func (w *WorkflowStep) SetNextStep(next *WorkflowStep) {
```

Comment:
Trying to step through code this is a bit complex to follow

Why are SetNextStep and NextStep unused. What would we consider a new step vs a substep

Reply:
Other than that I think it's pretty easy to follow nice.